corehq/apps/reports/views.py

```diff
@@ -180,7 +179,19 @@ def _inner(request, domain, *args, **kwargs):
 can_view_form_attachment = _can_view_form_attachment()
 
 
+def location_restricted_access(view_func):
+    def _inner(request, domain, *args, **kwargs):
+        location_restricted_user = not request.couch_user.has_permission(domain, 'access_all_locations')
+        if not location_restricted_user or (
+                location_restricted_user and toggles.LOCATION_RESTRICTED_SCHEDULED_REPORTS.enabled(domain)):
+            return view_func(request, domain, *args, **kwargs)
+        return no_permissions(request)
+    return _inner
+
+
 @login_and_domain_required
+@location_safe
+@location_restricted_access
```

Comment:
I think we would be better with using `conditionally_location_safe` as Ethan suggested to avoid messing with these views for domains that don't have the feature flag enabled at all.

Good to test this, by just writing code looking at the documentation of the function, the decorator would be
```
conditionally_location_safe(location_restricted_scheduled_reports_enabled)
```
and the function to be called would be
```
def location_restricted_scheduled_reports_enabled(request, domain, *args, **kwargs):
    return toggles.LOCATION_RESTRICTED_SCHEDULED_REPORTS.enabled(domain)
```

Reply:
https://github.com/dimagi/commcare-hq/pull/32964/commits/bfb9e6f588ca40409b059e35c00be04a990e6b7a

There was a slight change in how domain value is accessed